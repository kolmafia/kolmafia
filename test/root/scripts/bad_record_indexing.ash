record rec {
  int x;
};
rec a;
int ax1 = a.x;
int ax2 = a[0];
int ax3 = a[ax1];
rec b = new rec;
rec c = new rec(b.x);
rec d = new rec[1];
rec e = new rec[a.x];

print(rec[int]{1: new rec(1)}[3][0]);
print(rec[int]{
  2: new rec(2)
}[3][0]);

// Fine, but never printed, since the script is never evaluated.
print(rec[int]{0: new rec(0)}[3].x);
