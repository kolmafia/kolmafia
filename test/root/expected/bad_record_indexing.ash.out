'a' cannot be indexed (bad_record_indexing.ash, line 6, char 11 to char 13)
'a' cannot be indexed (bad_record_indexing.ash, line 7, char 11 to char 13)
'new rec' cannot be indexed (bad_record_indexing.ash, line 10, char 9 to char 17)
'new rec' cannot be indexed (bad_record_indexing.ash, line 11, char 9 to char 17)
Too many keys for '{1: new rec(1)}' (bad_record_indexing.ash, line 13, char 15 to char 34)
Too many keys for '{ ... }' (bad_record_indexing.ash, line 14, char 15 to line 16, char 6)
